src/core/lib/channel/metrics.h

```diff
@@ -0,0 +1,225 @@
+// Copyright 2024 The gRPC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_SRC_CORE_LIB_CHANNEL_METRICS_H
+#define GRPC_SRC_CORE_LIB_CHANNEL_METRICS_H
+
+#include <grpc/support/port_platform.h>
+
+#include <atomic>
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "absl/strings/string_view.h"
+
+#include <grpc/support/log.h>
+
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/gprpp/sync.h"
+
+namespace grpc_core {
+
+// Static only.
+class GlobalInstrumentsRegistry {
+ public:
+  enum class Type {
+    kUndefined,
+    kUInt64,
+    kDouble,
+  };
+  struct GlobalCounterDescriptor {
+    Type type;
+    absl::string_view name;
+    absl::string_view description;
+    absl::string_view unit;
+    std::vector<absl::string_view> label_keys;
+    std::vector<absl::string_view> optional_label_keys;
+  };
+  struct GlobalHistogramDescriptor {
+    Type type;
+    absl::string_view name;
+    absl::string_view description;
+    absl::string_view unit;
+    std::vector<absl::string_view> label_keys;
+    std::vector<absl::string_view> optional_label_keys;
+  };
+  struct GlobalHandle {
```

Comment:
I think we should make it clear that we are using the handle as an index in docs, and that that is an API. (unless, we want to hide the fact that handle is just an index)

Reply:
Added some comment about the index. I think we can expose it so that StatsPlugins can use it to uniquely identify an instrument later with the handle. PTAL.

Reply:
GlobalInstrumentHandle? I think `grpc_core::GlobalHandle` sounds very generic